### ISSUE TITLE
Share buttons don't do anything on devices without navigator.share

### ISSUE THREAD

**Issue by aeharding** [aeharding/voyager#183]
e.g. desktop computer Firefox

We should probably build our own hook, useShare (or something).

It will accept exact same props as navigator.share.

If navigator.share is undefined, should copy text to clipboard (prefer URL if exists in share payload) and then present toast with useIonToast.

**Comment by rsammelson** [comment]
I'll take a crack at this one as well, but it might be above my JS abilities. I'll update once I try it.

**Comment by Vishram1123** [comment]
Maybe this polyfill could be of use: https://www.npmjs.com/package/share-api-polyfill

**Comment by Vishram1123** [comment]
I just saw that you made an edit which would copy the post/image url to the clipboard if navigator.share isn't present, but I think the share-api-polyfill would still be much more comprehensive and useful.

**Comment by rsammelson** [comment]
At least in my opinion, on desktop copying to clipboard is much more useful. Then I can just Super+Tab to the window I want to put the link in and paste it.

**Comment by aeharding** [comment]
Yeah, I think just copy to clipboard with a toast is best as a fallback.
